services/core/shared/transactionStatistics.js

```diff
@@ -13,79 +13,122 @@
+		unorderedfinalResult[entry.amount_range] += entry.count;
+		// SELECT amount_range, sum(count) AS count FROM transaction_statistics
+		// 	GROUP BY amount_range
+	});
+
+	const orderedFinalResult = {};
+	Object.keys(unorderedfinalResult).sort().reverse()
+		.forEach(amountRange => orderedFinalResult[amountRange] = unorderedfinalResult[amountRange]);
+	// 	ORDER BY amount_range DESC`, transformParamsForDb(params));
+
+	return orderedFinalResult;
 };
 
 const getDistributionByType = async params => {
-	const result = await database.any(`SELECT type, sum(count) AS count FROM transaction_statistics
-		WHERE $<dateFrom> <= timestamp AND timestamp <= $<dateTo>
-		GROUP BY type
-		ORDER BY type ASC`, transformParamsForDb(params));
-	return result;
-};
+	const db = await getDbInstance(config.db.collections.transaction_statistics.name);
 
-const ensureDbTableIsCreated = async () => {
-	try {
-		const db = getDatabase();
-		const tableExists = await db.any('SELECT to_regclass($1)', dbQueries.tableName);
+	const result = await db.find({
+		selector: {
+			date: {
+				$gte: params.dateFrom.toISOString(),
+				$lte: params.dateTo.toISOString(),
+			},
+			// 	WHERE $<dateFrom> <= timestamp AND timestamp <= $<dateTo>
+		},
+	});
 
-		if (!tableExists[0].to_regclass) {
-			await db.any(dbQueries.createTable);
-			logger.debug(`Initialized postgres table: ${dbQueries.tableName}`);
-		}
-	} catch (error) {
-		logger.error(`Failed to initialize postgres table: ${dbQueries.tableName}. Error: ${error}`);
-	}
+	const unorderedfinalResult = {};
+	result.forEach(entry => {
+		if (!unorderedfinalResult[entry.type]) unorderedfinalResult[entry.type] = 0;
+		unorderedfinalResult[entry.type] += entry.count;
+		// SELECT type, sum(count) AS count FROM transaction_statistics
+		// 	GROUP BY type
+	});
+
+	const orderedFinalResult = {};
+	Object.keys(unorderedfinalResult).sort((a, b) => Number(a) - Number(b))
+		.forEach(type => orderedFinalResult[type] = unorderedfinalResult[type]);
+	// 	ORDER BY type ASC`, transformParamsForDb(params));
+
+	return orderedFinalResult;
 };
 
-const fetchTransactionsForPastNDays = n => {
-	const db = getDatabase();
+const fetchTransactionsForPastNDays = async n => {
+	const db = await getDbInstance(config.db.collections.transaction_statistics.name);
 	[...Array(n)].forEach(async (_, i) => {
 		const date = moment().subtract(i, 'day').utc().startOf('day')
-			.toDate();
-		const shouldUpdate = i === 0 || !(await db.oneOrNone(dbQueries.selectDay, { date }));
+			.toISOString();
+		const shouldUpdate = i === 0 || !((await db.findOneByProperty('date', date)).length);
 		if (shouldUpdate) {
 			transactionStatisticsQueue.add({ date });
 		}
 	});
 };
 
-const init = async historyLengthDays => {
-	await ensureDbTableIsCreated();
-	fetchTransactionsForPastNDays(historyLengthDays);
-};
+const init = async historyLengthDays => fetchTransactionsForPastNDays(historyLengthDays);
```

Comment:
is there any need to make it async?

Reply:
`fetchTransactionsForPastNDays` being an async function, it is better to mark the `init` and `updateTodayStats` methods async.

Reply:
`fetchTransactionsForPastNDays` is an async method. Added `await` keyword to make it explicit.